models/search-model.js

```diff
@@ -2,25 +2,34 @@ const database = require('../database/dbConfig');
 
 // Fetch posts, comments, users, rooms based on search input
 const getFullSearchResults = async (search) => {
+    // TODO: posts and comments should ideally come back with the authors display_name, in this context. 
+  // just need the right sql calls... 
   const posts = await database('posts')
+    .join('users', 'users.id', 'posts.user_id')
+    .select('users.display_name', 'posts.title', 'posts.description', 'posts.created_at')
     .whereRaw('LOWER(posts.title) LIKE ?', [`%${search}%`])
     .orWhereRaw('LOWER(posts.description) LIKE ?', [`%${search}%`])
     .where('posts.visible', 1);
+
   const comments = await database('comments')
+    .join('users', 'users.id', 'comments.user_id')
+    .select('users.display_name', 'comments.comment', 'comments.created_at')
     .whereRaw('LOWER(comments.comment) LIKE ?', [`%${search}%`])
-    .where('comments.visible', 1);
+    .where('comments.visible', 1)
+    // ? .join('users', 'users.id', 'comments.user_id');
```

Comment:
Do we still need this? Not sure what it's doing.

Reply:
i am not sure, to be frank! I just left it in because it was there when i got there. lets review it further next time we get together?

Reply:
upon further inspection, i believe this has to do with moderation (eg mods being able to take a post down) although I am not certain!

Reply:
OK!